engine/api/workflow_run.go

```diff
@@ -914,71 +914,71 @@ func (api *API) postWorkflowRunHandler() service.Handler {
 
 		// Workflow Run initialization
 		sdk.GoRoutine(context.Background(), fmt.Sprintf("api.initWorkflowRun-%d", lastRun.ID), func(ctx context.Context) {
-			api.initWorkflowRun(ctx, api.mustDB(), api.Cache, p, wf, lastRun, opts, c)
+			api.initWorkflowRun(ctx, p.Key, wf, lastRun, opts, c)
 		}, api.PanicDump())
 
 		return service.WriteJSON(w, lastRun, http.StatusAccepted)
 	}
 }
 
-func (api *API) initWorkflowRun(ctx context.Context, db *gorp.DbMap, cache cache.Store, p *sdk.Project, wf *sdk.Workflow,
-	wfRun *sdk.WorkflowRun, opts *sdk.WorkflowRunPostHandlerOption, u *sdk.AuthConsumer) {
+func (api *API) initWorkflowRun(ctx context.Context, projKey string, wf *sdk.Workflow, wfRun *sdk.WorkflowRun, opts *sdk.WorkflowRunPostHandlerOption, u *sdk.AuthConsumer) {
 	var asCodeInfosMsg []sdk.Message
 	report := new(workflow.ProcessorReport)
 	defer func() {
-		go WorkflowSendEvent(context.Background(), db, cache, p.Key, report)
+		go WorkflowSendEvent(context.Background(), api.mustDB(), api.Cache, projKey, report)
 	}()
 
-	// IF NEW WORKFLOW RUN
+	p, err := project.Load(api.mustDB(), api.Cache, projKey,
```

Comment:
why? not sure we want to have project with clear integration/key in all execution of workflow run.

Reply:
Copy specific load without clear integ from handler.